### ISSUE TITLE
Query documents containing only comments are not rejected by static validator

### ISSUE THREAD

**Issue by chrisbutcher** [rmosolgo/graphql-ruby#1149]
See https://github.com/rmosolgo/graphql-ruby/pull/1148

> Expected: The result should populate errors to indicate to the client that the request was invalid
Actual: The result contains only {}

As seen in GraphiQL (the client detects the query as invalid, but the gem code on the server-side still accepts it:
![image](https://user-images.githubusercontent.com/1916444/33391265-7601eae2-d506-11e7-8705-a4b6baafc577.png)

The reference implementation says its invalid:
```
{
  "errors": [
    {
      "message": "Syntax Error GraphQL request (1:22) Unexpected <EOF>\n\n1: # Welcome to GraphiQL\n                        ^\n",
      "locations": [
        {
          "line": 1,
          "column": 22
        }
      ]
    }
  ]
}
```

cc: @rmosolgo @swalkinshaw @eapache

**Comment by rmosolgo** [comment]
It looks like this "fix" was added a few years ago (https://github.com/rmosolgo/graphql-ruby/commit/a37b4a593c8f185c610584f8da5f83a42bec3e65), it's ok with me to change the behavior to match the JS implementation. 

Any objections?

**Comment by daemonsy** [comment]
Possibly will be resolved by https://github.com/rmosolgo/graphql-ruby/pull/2344

**Comment by rmosolgo** [comment]
Fixed by #2344
